zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkConnection.java

```diff
@@ -192,4 +234,59 @@ public String getServers() {
   public void addAuthInfo(String scheme, byte[] auth) {
     _zk.addAuthInfo(scheme, auth);
   }
+
+  private void lookupGetChildrenMethod() {
```

Comment:
Is this one time look up? Why we dont do it at initial stage and make it final?

Reply:
1. It is a two step lookup: lookup the client lib, and call server. If we do lookup at the construction stage, we have to call server to know if server supports pagination or not. This is an extra call. 
2.  We'd like to save this extra call by using lazy init.
a. only when the first getChildren call, we get the method from client zk, then send it to server
b. If server does not support, we fallback to non-paginated. And cache it.
c. Future calls, it’ll always use the cached method to getChildren

Synced with @dasahcc offline, he agreed.